Neos.Flow/Classes/Mvc/Controller/AbstractController.php

```diff
@@ -12,6 +12,8 @@
  */
 
 use GuzzleHttp\Psr7\Uri;
+use Neos\Flow\ResourceManagement\PersistentResource;
```

Comment:
This sucks, but is required unless someone else has a good idea how to trigger resource importing **after** validation but before controller invokation.